### PR TITLE
Add support for custom dollar amounts

### PR DESCRIPTION
**Pull request by jbuckner**
This allows a dollar amount array to be passed in to customize the values we show. Anywhere from 3-7 values can be laid out properly.

amount button layouts (1)

### CODE REVIEW

File: packages/donation-form/src/donation-form.ts [hunk 1]
```diff
@@ -55,16 +56,14 @@ export class DonationForm extends LitElement {
...
   @property({ type: Boolean }) private creditCardVisible = false;
 
   @property({ type: Boolean }) private contactFormVisible = false;
 
   @property({ type: Boolean }) private donationInfoValid = true;
 
-  @property({ type: Boolean }) private contactInfoValid = false;
-
-  @property({ type: Boolean }) private hostedFieldsValid = false;
-
```
Comment (bfalling) [inline packages/donation-form/src/donation-form.ts:67]: QUESTION: I assume this deletion is just some cleanup? (Not directly relevant to ticket?)

File: packages/edit-donation/src/donation-form-edit-donation.ts [hunk 2]
```diff
@@ -109,9 +111,27 @@ export class DonationFormEditDonation extends LitElement {
...
+      this.customAmountSelected = false;
+      this.updateSelectedDonationInfo();
+      this.setupAmountColumnsLayoutConfig();
+    }
     if (changedProperties.has('donationInfo')) {
       this.updateSelectedDonationInfo();
     }
+    if (
+      changedProperties.has('defaultSelectedAmount') &&
+      this.defaultSelectedAmount
```
Comment (bfalling) [inline packages/edit-donation/src/donation-form-edit-donation.ts:127]: QUESTION: Is there a distinction needed between a `defaultSelectedAmount` of `0` and some other false-y value?

**Review by bfalling — APPROVED** [review]
LGTM. A couple questions.

File: packages/edit-donation/src/donation-form-edit-donation.ts (hunk 2 above)
Comment (jbuckner) [reply packages/edit-donation/src/donation-form-edit-donation.ts:127]: Ah good question.. I suppose `0` could be a default selected value and then it wouldn't get selected 😅 I'll update that check

File: packages/donation-form/src/donation-form.ts (hunk 1 above)
Comment (jbuckner) [reply packages/donation-form/src/donation-form.ts:67]: Oh yeah these were unrelated.. just noticed they weren't being used and removed them.

**Pull request merged**